### PR TITLE
Add RegExpAssertion support

### PR DESCRIPTION
**Pull request by kidi**
Hello @agourlay 

I have done a modest contribution on your test api library with the support of regexp;

```
          Then assert body.path("name").regExpMatch(".*man".r)
```

let me know if you like it ^^

### CODE REVIEW

File: src/main/scala/com/github/agourlay/cornichon/json/JsonSteps.scala [hunk 1]
```diff
@@ -94,6 +96,18 @@ object JsonSteps {
       )
     }
 
+    def regExpMatch(expectedRegex: Regex) = {
+      val baseTitle = if (jsonPath == JsonPath.root) s"$target mathes '$expectedRegex'" else s"$target's field '$jsonPath' matches '$expectedRegex'"
```
Comment (agourlay) [inline src/main/scala/com/github/agourlay/cornichon/json/JsonSteps.scala]: typo ```matches```

File: src/main/scala/com/github/agourlay/cornichon/json/JsonSteps.scala [hunk 2]
```diff
@@ -94,6 +96,18 @@ object JsonSteps {
       )
     }
 
+    def regExpMatch(expectedRegex: Regex) = {
```
Comment (agourlay) [inline src/main/scala/com/github/agourlay/cornichon/json/JsonSteps.scala]: I thin I would prefer the name ```matchesRegex``` to fit the DSL.
WDYT?

File: src/test/scala/com/github/agourlay/cornichon/steps/regular/assertStep/StringAssertionSpec.scala [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import cats.scalatest.{ ValidatedMatchers, ValidatedValues }
+import com.github.agourlay.cornichon.util.Instances._
+import org.scalatest.{ Matchers, WordSpec }
+
+class StringAssertionSpec extends WordSpec
+    with Matchers
+    with ValidatedMatchers
+    with ValidatedValues {
+
+  "StringAssertion" must {
```
Comment (agourlay) [inline src/test/scala/com/github/agourlay/cornichon/steps/regular/assertStep/StringAssertionSpec.scala]: 👍

**Comment by agourlay** [comment]
Thank you for your contribution!

This looks good to me, I have no problem merging this once all comments are addressed.

**Comment by kidi** [comment]
Great ;)

**Pull request merged**